File: Gabriel_Cadena_FrontEnd/src/app/models/transaction.ts
export interface Transaction {
  id?: string;
  transactionDate?: Date;
  transactionType: string;
  productId: number;
  productName?: string;
  quantity: number;
  unitPrice: number;
  totalPrice?: number;
  details: string;
}

export interface TransactionList {
  data: Transaction[];
    pageNumber: number;
    pageSize: number;
    totalRecords: number;
    totalPages: number;
  }

File: Gabriel_Cadena_BackEnd/TransactionService/Models/Transaction.cs
namespace TransactionService.Models
{
    public enum TransactionType
    {
        Purchase,
        Sale
    }

    public class Transaction
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public DateTime TransactionDate { get; set; } = DateTime.UtcNow;
        public string TransactionType { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal TotalPrice { get; set; }
        public string Details { get; set; }
    }
}